src/js/dataEntry.js

```diff
@@ -0,0 +1,95 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.defaults("gpii.chartAuthoring.dataEntry", {
+        gradeNames: ["gpii.chartAuthoring.templateInjection", "gpii.chartAuthoring.valueBinding", "autoInit"],
+        selectors: {
+            input: ".gpiic-ca-dataEntry-value",
+            percentage: ".gpiic-ca-dataEntry-percentage",
+            label: ".gpiic-ca-dataEntry-label"
+        },
+        strings: {
+            inputPlaceholder: "Value",
+            percentage: "%percentage%"
+        },
+        model: {
+            // value: number
+            label: ""
+            // perecentage: number
+            // total: number
+        },
+        modelRelay: {
+            source: "",
+            target: "percentage",
+            singleTransform: {
+                type: "fluid.transforms.free",
+                args: [
+                    "{that}.model.value",
+                    "{that}.model.total"
+                ],
+                func: "gpii.chartAuthoring.dataEntry.calculatePercentage"
+            }
+        },
+        bindings: {
+            input: "value",
+            label: "label"
+        },
+        invokers: {
+            getPercentageToRender: {
+                funcName: "gpii.chartAuthoring.percentage.percentageIfValue",
+                args: ["{that}.model.percentage", "{that}.model.value", ""]
+            },
+            setPercentage: {
+                funcName: "gpii.chartAuthoring.percentage.render",
+                args: [
+                    "{that}.dom.percentage",
+                    {
+                        expander: {
+                            func: "{that}.getPercentageToRender"
+                        }
+                    },
+                    "{that}.options.strings.percentage",
+                    2
+                ],
+                dynamic: true
+            }
+        },
+        listeners: {
+            "onCreate.setPercentage": "{that}.setPercentage",
+            "onCreate.setInputAttrs": {
+                "this": "{that}.dom.input",
+                "method": "attr",
+                "args": {
+                    placeholder: "{that}.options.strings.inputPlaceholder"
+                }
+            }
+        },
+        modelListeners: {
+            "value": {
+                listener: "{that}.setPercentage",
+                excludeSource: "init"
+            },
+            "total": {
+                listener: "{that}.setPercentage",
+                excludeSource: "init"
+            }
+        }
+    });
+
+    gpii.chartAuthoring.dataEntry.calculatePercentage = function (value, total) {
```

Comment:
@jobara - ok - in that case my original advice should work. The standard idiom for a transform function is for it to not be activated if its input value is undefined, and so produce no output (which then also remains undefined if it originally was). This behaviour comes for free, implemented in the Model Transformation system itself, for a transform which declares the standardTransformFunction grade and the transform function itself doesn't need to take care of it.